implementation/commons/src/main/java/eu/efti/commons/dto/UilDto.java

```diff
@@ -1,23 +1,36 @@
 package eu.efti.commons.dto;
 
 import io.swagger.v3.oas.annotations.media.Schema;
-import jakarta.validation.Valid;
-import jakarta.validation.constraints.NotNull;
+import jakarta.validation.constraints.NotBlank;
+import jakarta.validation.constraints.Pattern;
+import jakarta.validation.constraints.Size;
 import lombok.AllArgsConstructor;
+import lombok.Builder;
 import lombok.Data;
-import lombok.EqualsAndHashCode;
 import lombok.NoArgsConstructor;
-import lombok.experimental.SuperBuilder;
 
 @Data
-@SuperBuilder
+@Builder
 @NoArgsConstructor
 @AllArgsConstructor
-@EqualsAndHashCode(callSuper = true)
-public class UilDto extends AbstractUilDto implements ValidableDto {
+public class UilDto implements ValidableDto {
+    private static final String REGEX_URI = "\\w+:(/?/?)[^\\s]+";
 
-    @Valid
-    @NotNull(message= "AUTHORITY_MISSING")
-    @Schema( example = "see AuthorityDto")
-    private AuthorityDto authority;
+    @NotBlank(message = "GATE_ID_MISSING")
+    @Size(max = 255, message = "GATE_ID_TOO_LONG")
+    @Pattern(regexp = REGEX_URI, message = "GATE_ID_INCORRECT_FORMAT")
+    @Schema(example = "regex = \\w+:(\\/?\\/?)[^\\s]+")
+    private String gateId;
+
+    @NotBlank(message = "DATASET_ID_MISSING")
+    @Size(max = 36, message = "DATASET_ID_TOO_LONG")
+    @Pattern(regexp = "[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-4[0-9a-fA-F]{3}-[89aAbB][0-9a-fA-F]{3}-[0-9a-fA-F]{12}", message = "DATASET_ID_INCORRECT_FORMAT")
+    @Schema(example = "regex = [0-9a-fA-F]{8}-[0-9a-fA-F]{4}-4[0-9a-fA-F]{3}-[89aAbB][0-9a-fA-F]{3}-[0-9a-fA-F]{12}")
+    private String datasetId;
+
+    @NotBlank(message = "PLATFORM_ID_MISSING")
+    @Size(max = 255, message = "PLATFORM_ID_TOO_LONG")
+    @Pattern(regexp = REGEX_URI, message = "PLATFORM_ID_INCORRECT_FORMAT")
```

Comment:
This should allow platform ids that are not URLs as it will most likely be in another format. I propose we require only that it is a max 255 character long string.

Reply:
As discussed yesterday this will be handled in another story : [https://efti4eu.atlassian.net/browse/EREF-55](https://github.com/EFTI4EU/reference-implementation/pull/url)